wremnants/unfolding_tools.py

```diff
@@ -7,42 +7,101 @@ def define_gen_level(df, gen_level, dataset_name, mode="wmass"):
-        elif mode == "wlike":
-            df = df.Define("postFSRmuons", "GenPart_status == 1 && (GenPart_statusFlags & 1) && GenPart_pdgId == 13")
-            df = df.Define("postFSRantimuons", "GenPart_status == 1 && (GenPart_statusFlags & 1) && GenPart_pdgId == -13")
-            df = df.Define("postFSRmuonIdx", "ROOT::VecOps::ArgMax(GenPart_pt[postFSRmuons])")
-            df = df.Define("postFSRantimuonIdx", "ROOT::VecOps::ArgMax(GenPart_pt[postFSRantimuons])")
-            df = df.Define("ptGen", "event % 2 == 0 ? GenPart_pt[postFSRmuons][postFSRmuonIdx] : GenPart_pt[postFSRantimuons][postFSRantimuonIdx]")
-            df = df.Define("etaGen", "event % 2 == 0 ? abs(GenPart_eta[postFSRmuons][postFSRmuonIdx]) : abs(GenPart_eta[postFSRantimuons][postFSRantimuonIdx])")    
+            if "Wplusmunu" in dataset_name:
+                idx = "postFSRantimuIdx" 
+                muons = "postFSRantimus"
+            else:
+                idx = "postFSRmuIdx" 
+                muons = "postFSRmus"
+
+            df = df.Define("ptGen", f"GenPart_pt[{muons}][{idx}]")
+            df = df.Define("absEtaGen", f"fabs(GenPart_eta[{muons}][{idx}])")                
+        else:
+            df = df.Define("muGen", "ROOT::Math::PtEtaPhiMVector(GenPart_pt[postFSRmus][postFSRmuIdx], GenPart_eta[postFSRmus][postFSRmuIdx], GenPart_phi[postFSRmus][postFSRmuIdx], GenPart_mass[postFSRmus][postFSRmuIdx])")
+            df = df.Define("antimuGen", "ROOT::Math::PtEtaPhiMVector(GenPart_pt[postFSRantimus][postFSRantimuIdx], GenPart_eta[postFSRantimus][postFSRantimuIdx], GenPart_phi[postFSRantimus][postFSRantimuIdx], GenPart_mass[postFSRantimus][postFSRantimuIdx])")
+            df = df.Define("VGen", "ROOT::Math::PxPyPzEVector(muGen)+ROOT::Math::PxPyPzEVector(antimuGen)")
+            df = df.Define("massVGen", "VGen.mass()")
+
+            if mode == "wlike":
+                df = df.Define("ptGen", "event % 2 == 0 ? GenPart_pt[postFSRmus][postFSRmuIdx] : GenPart_pt[postFSRantimus][postFSRantimuIdx]")
+                df = df.Define("absEtaGen", "event % 2 == 0 ? fabs(GenPart_eta[postFSRmus][postFSRmuIdx]) : fabs(GenPart_eta[postFSRantimus][postFSRantimuIdx])")    
+            elif mode == "dilepton":
+                df = df.Define("ptVGen", "VGen.pt()")
+                df = df.Define("absYVGen", "fabs(VGen.Rapidity())")  
     
     if mode == "wlike":
         df = df.Define("qGen", "event % 2 == 0 ? -1 : 1")
 
     return df
 
-        
+def define_fiducial_space(df, mode="wmass", pt_min=26, pt_max=55, mass_min=60, mass_max=120, mtw_min=40, selections=[]):
+    # Define a fiducial phase space where out of acceptance contribution is stored for additional cuts 
+    #   (those on the unfolding axes are not necessary)
+    
+    if mode == "wmass":
+        selection = f"(mTWGen > {mtw_min})"
+    elif mode == "wlike":
+        selection = f"""
+            (fabs(muGen.eta()) < 2.4) && (fabs(antimuGen.eta()) < 2.4) 
```

Comment:
Since most of the selection is in common between wlike and dilepton, I was wondering whether we should keep a common string and add pieces under each scope (currently only the traverse mass), to avoid mistakes when this gets updated in one field. Not super important though